interface/ceed-preconditioning.c

```diff
@@ -1961,6 +1961,105 @@ int CeedOperatorMultigridLevelCreateH1(CeedOperator op_fine, CeedVector p_mult_f
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Scale composite prolongation and restriction operators for multiplicity between suboperators.
+
+  Note: Calling this function asserts that setup is complete and sets the CeedOperator as immutable.
+
+  @param[in,out] op_prolong    Coarse to fine operator
+  @param[in,out] op_restrict   Fine to coarse operator
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref User
+**/
+int CeedCompositeOperatorScaleMultigridLevel(CeedOperator op_prolong, CeedOperator op_restrict) {
```

Comment:
I think the docs need to reference how these operators are created and their required relation.

Reply:
I think the extra information I added might clarify the use case here. This only works with suboperators created via `CeedOperatorMultigridLevelCreate*` functions. In that case, the scaling vector is never available to the user.

Reply:
My concern is that `CeedOperatorMultigridLevelCreate*` says
>   Note: Calling this function asserts that setup is complete and sets the `CeedOperator` as immutable.

and yet if they put it into a composite operator and call this function, their sub-operator will be mutated.

What if we did multigrid level creation directly for composite operators? Then the proper scaling would be done the first time.

Reply:
> What if we did multigrid level creation directly for composite operators? Then the proper scaling would be done the first time.

We could do that. We will need to decide what will happen with multiple active bases/restrictions and then pass an array of basis/restriction information. I thought this would be a quick fix, but I'll go tackle that piece of ugliness before trying to fix this bug again.

Reply:
I don't have a problem with a quick fix if we need to map out the problem space better, but we need to do it in a way that avoids expensive bug hunts in the future because this looks easy to misuse and we should avoid having false statements in documentation.

Reply:
The documentation is still true as written with the changes here - `CeedOperatorMultigridLevelCreate*` sets the input operator on the fine grid as immutable but does not mark the prolongation, restriction, or coarse grid operators as immutable. Perhaps it should though.

Reply:
Oh, I missed that distinction. I don't like the late mutation, but I guess we can live with it if a safer interface is too complicated.